File: 프로그래머스/lv2/42885. 구명보트/구명보트.js
function solution(people, limit) {
    
    people.sort((a,b)=>a-b);
    
    let cnt = 0;
    for(var l=0,r=people.length-1; l<r; r--){
        if((people[l] + people[r]) <= limit){
            cnt++;
            l++;
        }else{
            cnt++;
        }
    }
    if(l === r){
        cnt ++;
    }
    return cnt;
}

File: 프로그래머스/2/42885. 구명보트/구명보트.java
import java.util.Arrays;

class Solution {
    public int solution(int[] people, int limit) {
        Arrays.sort(people);

        int start = 0 ;
        int end = people.length-1;
        int answer = 0;

        while (start <= end) {
            int max = people[end];
            int min = people[start];

            if (max + min <= limit) {
                answer++;
                start++;
                end--;
            } else {
                answer++;
                end--;
            }
        }
        
        return answer;
    }
}